### PR TITLE
CSHARP-4541: Update libmongocrypt payloads to new QE protocol

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: bindings/cs/MongoDB.Libmongocrypt/Library.cs [hunk 1]
```diff
@@ -512,6 +515,10 @@ internal class Delegates
             /// uint64_t mongocrypt_crypt_shared_lib_version(const mongocrypt_t* crypt);
             /// </summary>
             public delegate ulong mongocrypt_crypt_shared_lib_version(MongoCryptSafeHandle handle);
+            /// <summary>
+            /// bool mongocrypt_setopt_fle2v2(mongocrypt_t *crypt, bool enable)
+            /// </summary>
+            public delegate bool mongocrypt_setopt_fle2v2(MongoCryptSafeHandle handle, bool enable);
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt/Library.cs:521]: I think you need to add this attribute: `[return: MarshalAs(UnmanagedType.I1)]`

File: bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs [hunk 2]
```diff
@@ -109,6 +109,7 @@ public static CryptClient Create(CryptOptions options)
                 }
 
                 Library.mongocrypt_setopt_use_need_kms_credentials_state(handle);
+                Library.mongocrypt_setopt_fle2v2(handle, true);
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs:112]: let's configure it as the first step after new as it's done in c impl: https://github.com/mongodb/mongo-c-driver/commit/8aced03a963e70182b79d883e881f239766b09bf#diff-6611c8b7be663c2fc9c7942692d314e8464f9398d0275e7ce691905fd5eeeeb0R1330

Comment (DmitryLukyanov) [reply bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs:112]: Also, you need to analyze method result, see `mongocrypt_setopt_encrypted_field_config_map` call above as example

**Review by DmitryLukyanov — CHANGES_REQUESTED** [review]

File: bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs:112]: I mean `handle.Check(` part

Comment (BorisDog) [reply bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs]: Done.

File: bindings/cs/MongoDB.Libmongocrypt/Library.cs (hunk 1 above)
Comment (BorisDog) [reply bindings/cs/MongoDB.Libmongocrypt/Library.cs:522]: Done.

**Review by DmitryLukyanov — APPROVED** [review]
Please ensure that new changes still work on the driver side and then LGTM

**Pull request merged**